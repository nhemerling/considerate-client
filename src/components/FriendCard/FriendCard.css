.FriendCard {
  background: none;
  display: flex;
  flex-direction: column;
  border: 1px solid #000;
  margin-bottom: 40px;
  margin-right: 20px;
  border-radius: 4px;
  width: 220px;
  padding: 20px;
}

.FriendCard__likes {
  list-style-type: none;
  margin: 0;
  padding: 0;
}
