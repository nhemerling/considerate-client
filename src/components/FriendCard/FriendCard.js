import React, { Component } from 'react';
import FriendApiService from '../../services/friend-api-service';
import moment from 'moment';
import './FriendCard.css';

export default class FriendCard extends Component {
  constructor(props) {
    super(props);
    this.state = {
      friendLikes: [],
    };
  }

  componentDidMount() {
    FriendApiService.getFriendLikes(this.props.friend.id).then((likes) => {
      this.setState({
        friendLikes: likes,
      });
    });
  }

  render() {
    const { friend } = this.props;
    const { friendLikes = [] } = this.state;
    return (
      <section className="FriendCard">
        <h2 className="FriendCard__name">{friend.friend_name}</h2>
        <h3 className="FriendCard__occasion">{friend.occasion}</h3>
        <p className="FriendCard__occasion-date">
          {moment(friend.occasion_date).format('MMMM Do YYYY')}
        </p>
        <ul className="FriendCard__likes">
          {friendLikes.map((like) => {
            return (
              <li key={like.id}>
                <p>{like.like_name}</p>
              </li>
            );
          })}
        </ul>
      </section>
    );
  }
}
