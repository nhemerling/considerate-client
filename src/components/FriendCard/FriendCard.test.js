import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import FriendCard from './FriendCard';

describe('FriendCard Component', () => {
  const friendProp = {
    id: 1,
    friend_name: 'Test name',
    occasion: 'Birthday',
    occasion_date: '2029-01-22T00:00:00.001Z',
  };

  it('renders a FriendCard with friend props', () => {
    const wrapper = shallow(<FriendCard friend={friendProp} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
