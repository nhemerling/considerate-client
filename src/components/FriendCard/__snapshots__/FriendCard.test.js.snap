// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FriendCard Component renders a FriendCard with friend props 1`] = `
<section
  className="FriendCard"
>
  <h2
    className="FriendCard__name"
  >
    Test name
  </h2>
  <h3
    className="FriendCard__occasion-date"
  >
    January 22nd 2029
  </h3>
  <h4
    className="FriendCard__occasion"
  >
    Birthday
  </h4>
  <ul
    className="FriendCard__likes"
  />
  <div
    className="FriendCard__buttons"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Delete
    </button>
    <Link
      to="/edit-friend/1"
    >
      <button
        type="button"
      >
        Edit
      </button>
    </Link>
  </div>
</section>
`;
